addons/web/static/tests/legacy/views/form/form_view_tests.js

```diff
@@ -14776,4 +14776,175 @@ QUnit.module("Views", (hooks) => {
+                    <group>
+                        <field name="display_name"/>
+                    </group>
+                </form>`,
+            resId: 1,
+            mockRPC(_, args) {
+                if (args.method === "web_save") {
+                    assert.step("save");
+                }
+            },
+        });
+
+        Object.defineProperty(document, 'visibilityState', {
+            value: 'hidden',
+        });
+        document.dispatchEvent(new Event("visibilitychange"));
+
+        await nextTick();
+        assert.verifySteps([], "should not have saved");
+    });
+
+    QUnit.test("Auto save: save on hiding tab (invalid field)", async function (assert) {
+            await makeView({
+            type: "form",
+            resModel: "partner",
+            serverData,
+            arch: `
+                <form>
+                    <group>
+                        <field name="display_name" required="1"/>
+                    </group>
+                </form>`,
+            resId: 1,
+            mockRPC(route, { args, method, model }) {
+                if (method === "web_save" && model === "partner") {
+                    assert.step("save"); // should not be called
+                }
+            },
+        });
+
+        await editInput(target, '.o_field_widget[name="display_name"] input', "");
+        Object.defineProperty(document, 'visibilityState', {
+            value: 'hidden',
+        });
+        document.dispatchEvent(new Event("visibilitychange"));
+        await nextTick();
+
+        assert.verifySteps([], "should not save because of invalid field");
+    });
+
+    QUnit.test("Auto save: save only once when hiding tab several times quickly", async (assert) => {
+        await makeView({
+            type: "form",
+            resModel: "partner",
+            serverData,
+            arch: `
+                <form>
+                    <group>
+                        <field name="display_name"/>
+                    </group>
+                </form>`,
+            resId: 1,
+            mockRPC(_, args) {
+                if (args.method === "web_save") {
+                    assert.step("save");
+                }
+            },
+        });
+
+        assert.notStrictEqual(
+            target.querySelector('.o_field_widget[name="display_name"]').value,
+            "test"
+        );
+
+        await editInput(target, '.o_field_widget[name="display_name"] input', "test");
+            
+        Object.defineProperty(document, 'visibilityState', {
+            value: 'hidden',
+        });
+        document.dispatchEvent(new Event("visibilitychange"));
+        document.dispatchEvent(new Event("visibilitychange"));
+        document.dispatchEvent(new Event("visibilitychange"));
+
+        await nextTick();
+        assert.verifySteps(["save"], "should have saved, but only once");
+    });
+
+    QUnit.test("Auto save: save only once when hiding tab and clicking save", async (assert) => {
```

Comment:
not sure this test is really relevant (also, you don't really assert what is saved anyway). What did you have in mind for this one? Did you observe a buggy situation that you wanted to reproduce?